app/models/host.rb

```diff
@@ -30,6 +35,19 @@ class Host < ApplicationRecord
   has_many :assigned_internal_profiles, -> { external(false) },
            through: :policies, source: :profiles
 
+  scope :with_benchmark, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+
+    joins(:policies).joins(
+      <<-SQL
+      INNER JOIN "profiles" ON "profiles"."policy_id" = "policies"."id" AND
+      "profiles"."os_minor_version" = ("hosts".#{OS_MINOR_VERSION})::text
```

Comment:
Can you please elaborate on the condition for OS minor version? 
It might not be necessary. When sorting by SSG version, it should sort by last test result's benchmark scoped by profile context.